dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEnrollmentAnalyticsTableManager.java

```diff
@@ -105,41 +124,29 @@ protected void populateTable( AnalyticsTableUpdateParams params, AnalyticsTableP
 
         List<AnalyticsTableColumn> columns = getDimensionColumns( program );
 
         validateDimensionColumns( columns );
 
         for ( AnalyticsTableColumn col : columns )
         {
-            sql += col.getName() + ",";
+            sql.append(col.getName()).append(",");
         }
 
-        sql = TextUtils.removeLastComma( sql ) + ") select ";
+        sql = new StringBuilder(TextUtils.removeLastComma(sql.toString()) + ") select ");
 
         for ( AnalyticsTableColumn col : columns )
         {
-            sql += col.getAlias() + ",";
+            sql.append(col.getAlias()).append(",");
         }
 
-        sql = TextUtils.removeLastComma( sql ) + " ";
-
-        sql += "from programinstance pi " +
-            "inner join program pr on pi.programid=pr.programid " +
-            "left join trackedentityinstance tei on pi.trackedentityinstanceid=tei.trackedentityinstanceid and tei.deleted is false " +
-            "inner join organisationunit ou on pi.organisationunitid=ou.organisationunitid " +
-            "left join _orgunitstructure ous on pi.organisationunitid=ous.organisationunitid " +
-            "left join _organisationunitgroupsetstructure ougs on pi.organisationunitid=ougs.organisationunitid " +
-                "and (cast(date_trunc('month', pi.enrollmentdate) as date)=ougs.startdate or ougs.startdate is null) " +
-            "left join _dateperiodstructure dps on cast(pi.enrollmentdate as date)=dps.dateperiod " +
-            "where pr.programid=" + program.getId() + " " +
-            "and pi.organisationunitid is not null " +
-            "and pi.lastupdated <= '" + getLongDateString( params.getStartTime() ) + "' " +
-            "and pi.incidentdate is not null " +
-            "and pi.deleted is false ";
-
-        invokeTimeAndLog( sql, String.format( "Populate %s", tableName ) );
+        sql = new StringBuilder(TextUtils.removeLastComma(sql.toString()) + " ");
+
+        sql.append("from programinstance pi " + "inner join program pr on pi.programid=pr.programid " + "left join trackedentityinstance tei on pi.trackedentityinstanceid=tei.trackedentityinstanceid and tei.deleted is false " + "inner join organisationunit ou on pi.organisationunitid=ou.organisationunitid " + "left join _orgunitstructure ous on pi.organisationunitid=ous.organisationunitid " + "left join _organisationunitgroupsetstructure ougs on pi.organisationunitid=ougs.organisationunitid " + "and (cast(date_trunc('month', pi.enrollmentdate) as date)=ougs.startdate or ougs.startdate is null) " + "left join _dateperiodstructure dps on cast(pi.enrollmentdate as date)=dps.dateperiod " + "where pr.programid=").append(program.getId()).append(" ").append("and pi.organisationunitid is not null ").append("and pi.lastupdated <= '").append(getLongDateString(params.getStartTime())).append("' ").append("and pi.incidentdate is not null ").append("and pi.deleted is false ");
```

Comment:
Could we revert this change - would prefer to have the string broken down line by line for readability - Java convert to string builder internally also.